bridge-impl/src/main/java/com/liferay/faces/bridge/context/internal/RenderRedirectWriterImpl.java

```diff
@@ -108,88 +108,118 @@ public Writer getWrapped() {
 		void invoke() throws IOException;
 	}
 
-	protected class CBufOffLenOutputOperation implements OutputOperation {
+	private static class CBufOffLenOutputOperation implements OutputOperation {
 
 		private char[] cbuf;
 		private int off;
 		private int len;
+		private Writer myWriter;
 
-		public CBufOffLenOutputOperation(char[] cbuf, int off, int len) {
+		public CBufOffLenOutputOperation(Writer writer, char[] cbuf, int off, int len) {
 			this.cbuf = cbuf.clone();
 			this.off = off;
 			this.len = len;
+			this.myWriter = writer;
 		}
+		
+//		public CBufOffLenOutputOperation(char[] cbuf, int off, int len) {
+//			this.cbuf = cbuf.clone();
+//			this.off = off;
+//			this.len = len;
+//		}
 
 		public void invoke() throws IOException {
-			wrappedWriter.write(cbuf, off, len);
+			myWriter.write(cbuf, off, len);
 		}
 	}
 
-	protected class CbufWriteOperation implements OutputOperation {
+	private static class CbufWriteOperation implements OutputOperation {
 
 		private char[] cbuf;
+		private Writer myWriter;
 
-		public CbufWriteOperation(char[] cbuf) {
+		public CbufWriteOperation(Writer writer, char[] cbuf) {
 			this.cbuf = cbuf.clone();
+			this.myWriter = writer;
 		}
 
 		public void invoke() throws IOException {
-			wrappedWriter.write(cbuf);
+			myWriter.write(cbuf);
 		}
 	}
 
-	protected class CloseOperation implements OutputOperation {
+	private static class CloseOperation implements OutputOperation {
+		
+		private Writer myWriter;
+		
+		public CloseOperation(Writer writer) {
+			this.myWriter = writer;
+		}
+		
 		public void invoke() throws IOException {
-			wrappedWriter.close();
+			myWriter.close();
 		}
 	}
 
-	protected class FlushOperation implements OutputOperation {
+	private static class FlushOperation implements OutputOperation {
+		
+		private Writer myWriter;
```

Comment:
@pcwhite Please rename "myWriter" to "writer" in RenderRedirectWriterImpl.java, squash your commits, and push to your branch again.